### PR TITLE
process gtps as soon they are applied

### LINKED ISSUE #221: [FEATURE]: Need GlobalTrafficPolicy config to applied when the event occurs
**Is your feature request related to a problem? Please describe.**
Currently, whenever GTP config is created/updated in service namespace, admiral's local cache gets updated and based on the observations made during manual testing, updates to Istio's destination rule will happen during the sync cycle. 
Currently, sync is executed every 10mins, so in worst cases, it will take 10 mins for GTP config to be in effect.

**Describe the solution you'd like**
Any CRUD operation on GTP should be as soon as the event occurs.

### PR DESCRIPTION
**Pull request by shriramsharma**
Fixes #221  process gtps as soon they are applied

Signed-off-by: Shriram Sharma <shriram_sharma@intuit.com>

### CODE REVIEW

File: admiral/pkg/clusters/types.go [hunk 1]
```diff
@@ -243,3 +247,19 @@ func HandleEventForDeployment(event admiral.EventType, obj *k8sAppsV1.Deployment
 	// Use the same function as added deployment function to update and put new service entry in place to replace old one
 	modifyServiceEntryForNewServiceOrPod(event, env, globalIdentifier, remoteRegistry)
 }
+
+// HandleEventForGlobalTrafficPolicy processes all the events related to GTPs
+func HandleEventForGlobalTrafficPolicy(event admiral.EventType, gtp *v1.GlobalTrafficPolicy, remoteRegistry *RemoteRegistry, clusterName string) {
```
Comment (ccfishk) [inline admiral/pkg/clusters/types.go:252]:
```suggestion
func HandleEventForGlobalTrafficPolicy(event admiral.EventType, gtp *v1.GlobalTrafficPolicy, remoteRegistry *RemoteRegistry, clusterName string) error {
```

Comment (ccfishk) [reply admiral/pkg/clusters/types.go:252]: wouldn't an error for error case would help the function ?

**Review by ccfishk — APPROVED** [review]

File: admiral/pkg/clusters/types.go (hunk 1 above)
Comment (nirvanagit) [inline admiral/pkg/clusters/types.go:252]: Can a few test cases be added for `HandleEventForGlobalTrafficPolicy`?

File: admiral/pkg/clusters/types.go [hunk 2]
```diff
@@ -180,14 +182,17 @@ func (dh *DependencyHandler) Deleted(obj *v1.Dependency) {
 
 func (gtp *GlobalTrafficHandler) Added(obj *v1.GlobalTrafficPolicy) {
 	log.Infof(LogFormat, "Added", "globaltrafficpolicy", obj.Name, gtp.ClusterID, "received")
+	HandleEventForGlobalTrafficPolicy(admiral.Add, obj, gtp.RemoteRegistry, gtp.ClusterID)
 }
 
 func (gtp *GlobalTrafficHandler) Updated(obj *v1.GlobalTrafficPolicy) {
 	log.Infof(LogFormat, "Updated", "globaltrafficpolicy", obj.Name, gtp.ClusterID, "received")
+	HandleEventForGlobalTrafficPolicy(admiral.Update, obj, gtp.RemoteRegistry, gtp.ClusterID)
```
Comment (nirvanagit) [inline admiral/pkg/clusters/types.go:190]: Will an assignment of `_ = HandleEventForGlobalTrafficPolicy(..)` help to prevent warning from Github Actions?

**Review by nirvanagit — APPROVED** [review]

**Pull request merged**